src/cc/bpf_module.cc

```diff
@@ -349,10 +348,13 @@ int BPFModule::annotate() {
       fn->addFnAttr(Attribute::AlwaysInline);
 
   // separate module to hold the reader functions
-  auto m = make_unique<Module>("sscanf", *ctx_);
+  auto m = std::make_unique<Module>("sscanf", *ctx_);
 
   size_t id = 0;
-  for (auto &table : *tables_) {
+  Path path({std::to_string((uintptr_t)this)});
+  for (auto it = ts_->lower_bound(path), up = ts_->upper_bound(path); it != up; ++it) {
+    TableDesc &table = it->second;
```

Comment:
Is it possible for us not to keep a copy of `TableDesc` pointers in `BPFModule` class? Users need this table related methods (like get name, ID, size etc.) can get a reference to the table storage class from `BPFModule` (or from their own reference if they passed it to `BPFModule` like the BPFFS case), and then call relevant method from the source-of-truth directly.

We can even keep the methods in `BPFModule` for a while for backward-compatible, but just proxy through `ts_->corresponding_method()`?